### DISCUSSION TITLE
Launch configuration setup commands explanation

### DISCUSSION THREAD

**Discussion by KarolisMil** [microsoft/vscode-cpptools#9665] [Q&A] [answered]
Hello,

Just out of curiosity.

Documentation on launch.json reference [here](https://code.visualstudio.com/docs/cpp/launch-json-reference) mentions these two:
`"setupCommands": [ { "text": "target-run", "description": "run target", "ignoreFailures": false }]`
`"customLaunchSetupCommands": [ { "text": "target-run", "description": "run target", "ignoreFailures": false }]`.

What is `target-run` command? I cannot find it in GDB/MI documentation, but can see that is being used in other people's examples.

My GDB also return an error:
```(gdb)
-target-run
^error,msg="Undefined MI command: target-run",code="undefined-command"
```

**Comment by sean-mcmanus** [comment]
@WardenGnaw ?

**Comment by WardenGnaw** [comment, marked as answer]
It seems to just be an example used from the original documentation. It may not be a real command since users will want to change it to be something they want to use to setup their program. 

See original document: https://github.com/microsoft/vscode-cpptools/blob/c0e32437df580fbc0c25f61f0b68b623ae3746dc/launch.md?plain=1#L109
